spec/end_to_end/with_partner_spec.rb

```diff
@@ -0,0 +1,101 @@
+      expect(content.dig("employments", 0, "payments", 0)).to eq({ "gross" => "1.0",
+                                                                   "tax" => "-0.0",
+                                                                   "national_insurance" => "-0.0",
+                                                                   "client_id" => "id-0",
+                                                                   "date" => "2023-02-15",
+                                                                   "benefits_in_kind" => 0,
+                                                                   "net_employment_income" => "1.0" })
+      expect(content["regular_transactions"]).to eq(
+        [{ "operation" => "credit", "category" => "friends_or_family", "frequency" => "weekly", "amount" => "200.0" }],
+      )
+      expect(content["state_benefits"].length).to eq 2
+      expect(content["capitals"]).to eq({
+        "bank_accounts" => [], "non_liquid_capital" => [{ "value" => "700.0", "description" => "Non Liquid Asset", "subject_matter_of_dispute" => false }]
+      })
+      expect(content["vehicles"]).to eq(
+        [{ "value" => "1.0", "loan_amount_outstanding" => 0, "date_of_purchase" => "2021-02-15", "in_regular_use" => false, "subject_matter_of_dispute" => false }],
+      )
+      expect(content["dependants"]).to eq(
+        [{ "date_of_birth" => "2012-02-15", "in_full_time_education" => true, "relationship" => "child_relative", "monthly_income" => 0, "assets_value" => 0 }],
+      )
+    end
+
+    get_assessment_stub = stub_request(:get, %r{assessments/assessment_id\z}).to_return(
+      body: FactoryBot.build(:api_result, eligible: "eligible").to_json,
+      headers: { "Content-Type" => "application/json" },
+    )
+
+    start_assessment
+    fill_in_provider_screen
+    fill_in_applicant_screen(partner: "Yes")
+    fill_in_dependants_screen
+    fill_in_client_income_screens
+    fill_in_client_capital_screens
+    fill_in_partner_details_screen(employed: "Employed and in work")
+    fill_in_partner_dependants_screen(child_dependants: "Yes", child_dependants_count: 1)
+    fill_in_partner_employment_screen
+    fill_in_partner_housing_benefit_screen(choice: "Yes")
+    fill_in_partner_housing_benefit_details_screen
+    fill_in_partner_benefits_screen(choice: "Yes")
+    fill_in_add_partner_benefit_screen
+    fill_in_partner_benefits_screen
+    fill_in_partner_other_income_screen(values: { friends_or_family: "200", other: "100" }, frequencies: { friends_or_family: "Every week" })
+    fill_in_partner_outgoings_screen
+    fill_in_partner_property_screen(choice: "Yes, with a mortgage or loan")
+    fill_in_partner_property_details_screen
+    fill_in_partner_vehicle_screen(choice: "Yes")
+    fill_in_partner_vehicle_details_screen
+    fill_in_partner_assets_screen(values: { valuables: "700" })
+    click_on "Submit"
```

Comment:
Yikes - this looks pretty hard to write a new test - it seems to be a backward step from the syntax we had before where only the 'interesting' screens needed to be specified in the test

Reply:
@starswan I would argue that here, in these end to end and flow tests (of which there are very few), we are explicitly testing the order of the screens. That's what these tests are about. Whereas in the new form tests, which replace the bulk of the old feature tests, we are now entirely free from implicit assumptions about flow (on the `alt-partner` branch it's proving to be an absolute nightmare to swap around the order of the screens because tests in completely unrelated feature tests keep breaking).

Reply:
I've now updated these tests so that, in the cases where the test is not about explicitly asserting the screens that are filled in, we use a new `fill_in_forms_until(:target_screen_name)` method so that we can skip to the interesting bits.